test/bind.spec.js

```diff
@@ -228,3 +228,19 @@ test('binding supports short syntax', async () => {
 
     expect(document.querySelector('span').classList.contains('bar')).toBeTruthy()
 })
+
+test('checkbox values are set correctly', async () => {
+    document.body.innerHTML = `
+        <div x-data="{ stringValue: 'foo', trueValue: true, falseValue: false }">
+            <input type="checkbox" name="stringCheckbox" :value="stringValue" />
+            <input type="checkbox" name="trueCheckbox" :value="trueValue" />
+            <input type="checkbox" name="falseCheckbox" :value="falseValue" />
+        </div>
+    `
+
+    Alpine.start()
+
+    expect(document.querySelector('input[name="trueCheckbox"]').value).toEqual('on')
+    expect(document.querySelector('input[name="falseCheckbox"]').value).toEqual('on')
```

Comment:
Not sure I understand this.

So if we bind false to the checkbox's value, we expect it to equal "on"?

Reply:
Yes, this is how it works in plain HTML. The value is `on` but it would not have a `checked` flag. If you submit the form to a server the browser will only send the values for checkboxes that are checked and omit those that aren‘t.